src/kibana/components/vislib/visualizations/time_marker.js

```diff
@@ -0,0 +1,58 @@
+define(function (require) {
+  var datemath = require('utils/datemath');
+
+  return function TimeMarkerFactory(d3) {
+    function TimeMarker(times, xScale, height) {
+      if (!(this instanceof TimeMarker)) {
+        return new TimeMarker(times, xScale, height);
+      }
+      var currentTimeArr = [new Date().getTime()];
+
+      this.xScale = xScale;
+      this.height = height;
+      this.lineClass = 'time-marker';
+      this.times = (times.length) ? times.map(function (dateMathString) {
+        return datemath.parse(dateMathString);
+      }) : currentTimeArr;
+    }
+
+    TimeMarker.prototype._isTimeBasedChart = function (selection) {
+      var data = selection.data();
+      return data.every(function (datum) {
+        return (datum.ordered && datum.ordered.date);
+      });
+    };
+
+    TimeMarker.prototype.render = function (selection) {
+      var self = this;
+
+      // return if not time based chart
+      if (!self._isTimeBasedChart(selection)) return;
+
+      selection.each(function () {
+        d3.select(this).selectAll('time-marker')
+          .data(self.times)
+          .enter().append('line')
+          .attr('class', self.lineClass)
+          .attr('x1', function (d) {
+            return self.xScale(d);
+          })
+          .attr('x2', function (d) {
+            return self.xScale(d);
+          })
+          .attr('y1', self.height)
+          .attr('y2', self.xScale.range()[0]);
+      });
+    };
+
+    TimeMarker.prototype.get = function (field) {
```

Comment:
A generic getter isn't useful here - all those properties are available on the instance, and it doesn't prevent needing outside knowledge of internal variable names. If you need a specific getter, make a method for it. It looks like this is mostly used for testing though, so can be removed.